keyless/pepper/service/src/groth16_vk.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::watcher::{unhexlify_api_bytes, ExternalResource};
+use anyhow::{anyhow, Result};
+use aptos_infallible::RwLock;
+use ark_bn254::{Bn254, G1Affine, G2Affine};
+use ark_groth16::{PreparedVerifyingKey, VerifyingKey};
+use ark_serialize::CanonicalDeserialize;
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
+pub struct VKeyData {
```

Comment:
For some 0x1 structs, there are structs already declared in `api/types/src`, but I guess not for everything, right?

Reply:
right, those don't help here.